pep-0501.txt

```diff
@@ -1,27 +1,29 @@
 PEP: 501
-Title: General purpose string interpolation
+Title: General purpose string template literals
 Version: $Revision$
 Last-Modified: $Date$
-Author: Nick Coghlan <ncoghlan@gmail.com>
-Status: Deferred
+Author: Nick Coghlan <ncoghlan@gmail.com>, Nick Humrich <nick@humrich.us>
+Status: Draft
 Type: Standards Track
 Content-Type: text/x-rst
-Requires: 498
+Requires: 498, 675, 701
 Created: 08-Aug-2015
-Python-Version: 3.6
+Python-Version: 3.12
 Post-History: 08-Aug-2015, 23-Aug-2015, 30-Aug-2015
 
 Abstract
 ========
```

Comment:
In general, the abstract is intended to be a concise, focused and easy to understand prose summary of what the PEP proposes (and is used by the RSS feeds and downstream tools/platforms as such), and both it and the PEP proper should be self-contained and be usable without directly relying on one another. Conversely, non-trivial background material explaining the issues with the current situation generally belongs in the Motivation section (which is missing here), while an explanation of how the PEP addresses the problem goes in the Rationale and/or Specification section, and examples belong in the spec (if brief/normative) or in a later section.

As it stands, I've been really struggling to understand from the abstract what the PEP is actually proposing, as it is in large part devoted to background material/motivation (which the PEP proper doesn't actually address) while including only a relatively brief mention of the proposed solution that didn't actually explain the basic purpose/function of template strings nor state how it actually solved the described problem, and followed by a list of examples that likewise weren't really helpful to me out of context without an appropriate explanation of what they were intended to accomplish.

Therefore, after applying the other suggested changes, I recommend you:

* Move the first few paragraphs, up until/not including that beginning "To address that problem", to a dedicated `Motivation` section (i.e. before the Proposal section and after the Summary of differences)
* Remove the examples, which are addressed later in the text where they are more appropriate and have sufficient prior explanation to be helpful to the reader
* Rewrite the remaining "To address this problem paragraph" accordingly, and to actually state the purpose and basic proposed behavior of template strings, along the lines of the following:

```rst
Though easy and elegant to use, Python :term:`f-string`\s
can be vulnerable to injection attacks when used to construct
shell commands, SQL queries, HTML snippets and similar
(for example, ``os.system(f"echo {message_from_user}")``).
This PEP introduces template literal strings (or "t-strings"),
which have the same syntax and semantics but with rendering deferred
until :func:`format` or another builder function is called on them.
This will allow standard library calls, helper functions
and third party tools to safety and intelligently perform
appropriate escaping and other string processing on inputs
while retaining same the usability and convenience of f-strings.
```

Reply:
>  (typo in the last line, though - same doesn't need to be there, I'm guessing it is left over from an earlier iteration of the final sentence)

Thanks, fixed above—as you inferred, it was an editing error

> put a `Background` header between the new abstract and the previous abstract

Or more idiomatically and conventionally per PEP 1 and PEP 12, a `Motivation` section as I suggest above.